### PR TITLE
Update SLEEF package.py

### PR DESCRIPTION
**Pull request by blapie**
Update preferred package to SLEEF 3.6

### CODE REVIEW

File: var/spack/repos/builtin/packages/sleef/package.py [hunk 1]
```diff
@@ -10,17 +10,20 @@ class Sleef(CMakePackage):
...
     license("BSL-1.0")
 
     version("master", branch="master")
+    version("3.6", commit="a99491afee2bae0b11e9ffbf3211349f43a5fd10", preferred=True)
     version("3.5.1_2020-12-22", commit="e0a003ee838b75d11763aa9c3ef17bf71a725bff")  # py-torch@1.8:
     version(
         "3.5.1",
         sha256="415ee9b1bcc5816989d3d4d92afd0cd3f9ee89cbd5a33eb008e69751e40438ab",
-        preferred=True,
+        deprecated=True,
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/sleef/package.py]: Why are we deprecating this version?

Comment (blapie) [reply var/spack/repos/builtin/packages/sleef/package.py]: For no particular reason, I'm happy to remove this line. I was not sure what rule was used to decide whether to deprecate or not.

Comment (alalazo) [reply var/spack/repos/builtin/packages/sleef/package.py]: Yeah, please do. Otherwise this LGTM.

Comment (blapie) [reply var/spack/repos/builtin/packages/sleef/package.py]: Done!

**Comment by blapie** [comment]
@alalazo Would you mind giving this a review please?

**Review by alalazo — APPROVED** [review]

**Comment by blapie** [comment]
Thank you!

**Pull request merged**